### PR TITLE
ISSUE-7076 Fix NPE in MQTT Connect

### LINKED ISSUE #7076: MQTT NPE if password or willMessage is null.
### Expected behavior
methods should return null not NPE

### Actual behavior
throws NPE

### Steps to reproduce
Send MQTT connect message, with a null password, and then call .password or .willMesage
https://github.com/netty/netty/pull/6817
This is caused by the following change where null byte[] for these two fields is not being null handled.

Work around is to use new methods and null check manually, but would be good to save random bugs to fix the changed logic in deprecated methods.

      String password = connect.payload().passwordInBytes() == null ? null : new String(connect.payload().passwordInBytes(), CharsetUtil.UTF_8);
      String willMessage = connect.payload().willMessageInBytes() == null ? null : new String(connect.payload().willMessageInBytes(), CharsetUtil.UTF_8);

### Netty version
4.1.13

### OS version (e.g. `uname -a`)

### PR DESCRIPTION
**Pull request by michaelandrepearce**
Add null checks before converting to string to avoid NPE.

Motivation:

Fixing an introduced NPE by #6817

Modification:

Add null checks.

Result:

Fixes #7076 .

### CODE REVIEW

**Comment by normanmaurer** [comment]
@michaelandrepearce can you please sign our ICLA and let me know once done:

https://netty.io/s/icla

Thanks!

**Comment by normanmaurer** [comment]
@michaelandrepearce also would be nice if you could just add some simple unit tests for this :)

**Comment by michaelandrepearce** [comment]
Already signed please check

File: codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java]: @michaelandrepearce please change to 2017

File: codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        Assert.assertNull(mqttConnectPayload.willMessageInBytes());
+        Assert.assertNull(mqttConnectPayload.willMessage());
+
+        builder = new MqttMessageBuilders.ConnectBuilder();
+        builder.willMessage((byte[]) null);
+
+        mqttConnectPayload = builder.build().payload();
+
+        Assert.assertNull(mqttConnectPayload.willMessageInBytes());
```
Comment (normanmaurer) [inline codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java]: consider add static import for `Assert.assertNull`

File: codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java (hunk 1 above)
Comment (michaelandrepearce) [inline codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java]: done.

File: codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java (hunk 2 above)
Comment (michaelandrepearce) [inline codec-mqtt/src/test/java/io/netty/handler/codec/mqtt/MqttConnectPayloadTest.java]: Done.

**Comment by michaelandrepearce** [comment]
Have just pushed both.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (1ba592049c).

@michaelandrepearce thanks!

**Pull request closed without merging**